site/components/pages/Retirements/SingleRetirement/ViewPledgeButton/index.tsx

```diff
@@ -0,0 +1,34 @@
+import { ButtonPrimary } from "@klimadao/lib/components";
+import { urls } from "@klimadao/lib/constants";
+import { t } from "@lingui/macro";
+import { Pledge } from "components/pages/Pledge/types";
+import { FC } from "react";
+
+type ViewPledgeButtonProps = {
+  pledge: Pledge | null;
+};
+
+/** Logic for hiding or showing a link to the users pledge depending on a valid pledge */
+export const ViewPledgeButton: FC<ViewPledgeButtonProps> = (props) => {
+  if (props.pledge) {
+    return (
+      <ButtonPrimary
+        href={`${urls.pledges}/${props.pledge?.ownerAddress}`}
```

Comment:
pass in renderLink here so that we can take advantage of nextjs pre-loading and client-side navigation

Reply:
Sure thing, updated. If there's no use case for the non-preloaded use of links should we remove it as an option on `BaseButton`? Thoughts?

Reply:
Sometimes we have strong CTA buttons that actually link to a page in a new tab (carbon calculators, or when bouncing between `site` and `app`), so for those we can't use `renderLink`